vip-jetpack/connection-pilot/class-jetpack-connection-attendant.php

```diff
@@ -0,0 +1,309 @@
+		add_filter( 'wp_is_application_passwords_available_for_user', [ $this, 'disable_application_passwords' ], 15, 2 );
+		add_filter( 'map_meta_cap', [ $this, 'modify_user_capabilties' ], PHP_INT_MAX, 4 );
+		add_filter( 'wpcom_vip_is_two_factor_forced', [ $this, 'bypass_two_factor_auth' ], PHP_INT_MAX );
+		add_action( 'plugins_loaded', [ $this, 'remove_my_jetpack_page' ], 30 );
+	}
+
+	/*
+	|--------------------------------------------------------------------------
+	| Owner Configuration
+	|--------------------------------------------------------------------------
+	*/
+
+	/**
+	 * Check if the current Jetpack connection owner is the VIP user.
+	 *
+	 * @return string 'is_vip' | 'is_legacy_vip' | 'not_vip'
+	 */
+	public function check_connection_owner_validity() {
+		$expected_owner_details = $this->get_expected_owner_details();
+		$legacy_owner_details   = $this->get_legacy_owner_details();
+		$org_owner_details      = $this->get_org_owner_details();
+
+		$current_owner = $this->get_current_connection_owner();
+		if ( ! isset( $current_owner->user_login ) ) {
+			return 'not_vip';
+		}
+
+		// We won't consider it the legacy user if it is still the expected owner.
+		if ( $expected_owner_details['login'] === $current_owner->user_login ) {
+			return 'is_vip';
+		}
+
+		if ( $legacy_owner_details && $legacy_owner_details['login'] === $current_owner->user_login ) {
+			return 'is_legacy_vip';
+		}
+
+		return 'not_vip';
+	}
+
+	/**
+	 * Returns the user object of the connection owner.
+	 *
+	 * @return WP_User|null Null if no connection owner found.
+	 */
+	private function get_current_connection_owner() {
+		if ( ! method_exists( 'Jetpack', 'connection' ) ) {
+			return null;
+		}
+
+		$jp_connection = \Jetpack::connection();
+
+		if ( ! method_exists( $jp_connection, 'get_connection_owner' ) ) {
+			return null;
+		}
+
+		$connection_owner = $jp_connection->get_connection_owner();
+		if ( false === $connection_owner ) {
+			return null;
+		}
+
+		return $connection_owner;
+	}
+
+	/**
+	 * Ensures the VIP user exists on the site and has the necessary permissions.
+	 *
+	 * @return WP_User|WP_Error WP_User if successful, WP_Error otherwise.
+	 */
+	public function ensure_user_exists() {
+		$details = $this->get_expected_owner_details();
+
+		$user = get_user_by( 'login', $details['login'] );
+
+		// Create user if it doesn't exist yet.
+		if ( ! $user ) {
+			$result = wp_insert_user( [
+				'user_login'   => $details['login'],
+				'user_email'   => $details['email'],
+				'display_name' => $details['display_name'],
+				'role'         => $details['role'],
+				'user_pass'    => wp_generate_password( 100 ),
+			] );
+
+			if ( is_wp_error( $result ) ) {
+				return new WP_Error( 'jp-cxn-pilot-owner-creation-failed', 'Failed to create new user.' );
+			}
+
+			$user = get_userdata( $result );
+			if ( ! $user ) {
+				return new WP_Error( 'jp-cxn-pilot-owner-creation-failed', 'Failed to create new user.' );
+			}
```

Comment:
It feels that the  second call (`get_userdata`) is a bit redundant? Do we expect some filters being in play? Cause otherwise we can just use the returned user id and the rest of the data from the original payload.

Reply:
`wp_insert_user()` only gives back an ID on success, whereas the rest of the function wants to utilize the user object (for `$user->user_email`, but mainly `$user->roles`).

It is nice as an extra check to make sure filters didn't break the user creation though.